content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md

```diff
@@ -1624,6 +1624,141 @@ my-new-cron-object   * * * * *   1          7s
 The `NAME` column is implicit and does not need to be defined in the CustomResourceDefinition.
 {{< /note >}}
 
+### Field Selectors
+
+[Field Selectors](/docs/concepts/overview/working-with-objects/field-selectors/)
+let clients select custom resources based on the value of one or more resource
+fields.
+
+All custom resources support the `metadata.name` and `metadata.namespace` field
+selectors.
+
+Fields declared in a `CustomResourceDefinition` may also be used with field
+selectors when `selectableFields` is used.
+
+#### SelectableFields
+
+{{< feature-state state="alpha" for_k8s_version="v1.30" >}}
+
+You need to enable the `CustomResourceFieldSelectors`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to
+use this behavior, which then applies to all CustomResourceDefinitions in your
+cluster.
+
+The `selectableFields` field of `CustomResourceDefinition` may be used to
+declare which other fields in a custom resource may be used in field selectors.
+The following example adds the `.spec.color` and `.spec.size` fields as
+selectable fields.
+
+Save the CustomResourceDefinition to `shirt-resource-definition.yaml`:
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: shirt.stable.example.com
+spec:
+  group: stable.example.com
+  scope: Namespaced
+  names:
+    plural: shirts
+    singular: shirt
+    kind: Shirt
+  versions:
+  - name: v1
+    served: true
+    storage: true
+    schema:
+      openAPIV3Schema:
+        type: object
+        properties:
+          spec:
+            type: object
+            properties:
+              color:
+                type: string
+              size:
+                type: string
+      selectableFields:
+      - jsonPath: .spec.color
+      - jsonPath: .spec.size
+      additionalPrinterColumns:
+      - jsonPath: .spec.color
+        name: Color
+        type: string
+      - jsonPath: .spec.size
+        name: Size
+        type: string
+```
+
+Create the CustomResourceDefinition:
+
+```shell
+kubectl apply -f shirt-resource-definition.yaml
+```
+
+Save some custom resources to `shirt-resources.yaml`:
+
+```yaml
+apiVersion: stable.example.com/v1
```

Comment:
```suggestion
---
apiVersion: stable.example.com/v1
```

When there's more than one document, I like to start each YAML document with a marker line (even the first one).